orderer/common/cluster/connectionsmgr.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package cluster
+
+import (
+	"sync"
+
+	"github.com/hyperledger/fabric/internal/pkg/comm"
+	"google.golang.org/grpc"
+)
+
+type ConnectionsCache map[string]*grpc.ClientConn
+
+func (cbc ConnectionsCache) Lookup(key string) (*grpc.ClientConn, bool) {
+	conn, ok := cbc[key]
+	return conn, ok
+}
+
+func (cbc ConnectionsCache) Put(key string, conn *grpc.ClientConn) {
+	cbc[key] = conn
+}
+
+func (cbc ConnectionsCache) Remove(key string) {
+	delete(cbc, key)
+}
+
+func (cbc ConnectionsCache) Size() int {
+	return len(cbc)
+}
+
+type ConnectionsMgr struct {
+	lock        sync.RWMutex
+	Connections ConnectionsCache
+	dialer      comm.ClientConfig
+}
+
+func (c *ConnectionsMgr) Connect(endpoint string) (*grpc.ClientConn, error) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	conn, alreadyConnected := c.Connections.Lookup(endpoint)
+	if alreadyConnected {
+		return conn, nil
+	}
+
+	conn, err := c.dialer.Dial(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	c.Connections.Put(endpoint, conn)
+
+	return conn, nil
+}
+
+func (c *ConnectionsMgr) Disconnect(endpoint string) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	conn, connected := c.Connections.Lookup(endpoint)
+	if !connected {
+		return
+	}
+	conn.Close()
+	c.Connections.Remove(endpoint)
+}
+
+func NewConnectionMgr(dialer comm.ClientConfig) *ConnectionsMgr {
```

Comment:
how do you update the TLS root certs in case of a config update?

Reply:
It will be updated as part of the existing Configure function - `func (ac *AuthCommMgr) Configure(channel string, members []RemoteNode) {` . 

```
// RemoteNode represents a cluster member
type RemoteNode struct {
	NodeAddress
	NodeCerts
}
type NodeCerts struct {
	// ServerTLSCert is the DER encoded TLS server certificate of the node
	ServerTLSCert []byte
	// ClientTLSCert is the DER encoded TLS client certificate of the node
	ClientTLSCert []byte
	// PEM-encoded X509 certificate authority to verify server certificates    <<<<<<< added this new field
	ServerRootCA []byte
}
```